### PR TITLE
refactor Legend to use withHandlers

### PR DESCRIPTION
**Pull request by mirecmrozek**

### CODE REVIEW

File: client/src/components/Public/Legend/Legend.js [hunk 1]
```diff
@@ -57,10 +55,14 @@ const Legend = ({legendOpen, setLegendOpen}: Props) => (
...
       Legenda
     </button>
   )
-)
 
-export default withState('legendOpen', 'setLegendOpen', true)(Legend)
+export default compose(
+  withState('legendOpen', 'setLegendOpen', true),
+  withHandlers({
+    toggleLegend: ({setLegendOpen}) => (e) => setLegendOpen((current) => !current),
```
Comment (borisvida) [inline client/src/components/Public/Legend/Legend.js]: The param `e` isn't used.

File: client/src/components/Public/Legend/Legend.js [hunk 2]
```diff
@@ -2,18 +2,18 @@
...
-import {withState} from 'recompose'
+import {compose, withState, withHandlers} from 'recompose'
 import {FACEBOOK_LIKE_SRC} from '../../../constants'
 import './Legend.css'
 import MapIcon from '../../../assets/mapIcon.svg'
 
 type Props = {
   legendOpen: boolean,
-  setLegendOpen: (open: boolean) => void,
+  toggleLegend: (open: boolean) => void,
```
Comment (borisvida) [inline client/src/components/Public/Legend/Legend.js]: This type is wrong IMHO.

**Comment by mirecmrozek** [comment]
Should be fine now.

**Comment by mpinter** [comment]
Looks good.

**Pull request merged**